### DISCUSSION TITLE
Is it possible to support multiple test sets in the future?

### DISCUSSION THREAD

**Discussion by prttt26** [lululxvi/deepxde#1800] [Q&A]
I want to monitor the model accuracy trend with iterations on an extended data domain from the training set, and comparing it with the original test set (randomly chosen from the same domain of the training set). (For example, the training and original test set comes from x in (0,1), and we want to see its performance for x in (1,2))
To do so, multiple test sets seems to be the most convenient method, but it's not supported now. 
If I want to apply this function by myself, it seems I need to write my own data and model, is that correct? 
(Or support of unrelated test like in https://github.com/lululxvi/deepxde/issues/1796 would also be helpful to my aim)

**Comment by lululxvi** [comment]
This can be implemented via a callback https://deepxde.readthedocs.io/en/latest/modules/deepxde.html#deepxde.callbacks.Callback
